### ISSUE TITLE
E501 reported in spite of noqa for string literal containing spaces

### ISSUE THREAD

**Issue by cjfuller** [charliermarsh/ruff#252]
The following code incorrectly reports an E501:

test.py
```python
"""
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaa
"""  # noqa: E501

```

```
$ ruff test.py
Found 1 error(s).
test.py:2:89: E501 Line too long (89 > 88 characters)
```

However without the space (and an extra "a" added to keep the length the same), this error is correctly ignored (no errors are reported:

```python
"""
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
"""  # noqa: E501

```

In both cases, flake8 does not report a line length error.

(ruff 0.0.44, python 3.8.14, ubuntu 22.04)

**Comment by charliermarsh** [comment]
(I think the issue here is that Flake8 respects the ` # noqa: E501` on the following line, while ruff only respects noqa directives on the same line.)

**Comment by cjfuller** [comment]
Sorry, why does the second example not error then? I initially thought that might be the case, but I was able to add an arbitrary number of lines to this string, and as long as none of them contained spaces, ruff was fine.

**Comment by charliermarsh** [comment]
Flake8 has the same behavior (unless I'm mistaken), which is: if you have a long line consisting of a single token, don't raise a line length error. This is helpful for things like URLs.

**Comment by charliermarsh** [comment]
I'll try to fix this today, I think it's important.
